static/flow/views/program-editor-panel.js

```diff
@@ -929,34 +927,46 @@ var ProgramEditorPanel = function(options) {
     //
     this.nameHash = {};
 
+    //
```

Comment:
@dstrawberrygirl This new updateNameHash function is used to keep the nameHash object up to date.  nameHash  contains a list of block objects which are referenced by block names - this function rebuilds the nameHash object so it is in sync with the current flow diagram.  This function is called when a program is loaded, block is deleted, block is renamed, or undo function is called.